### ISSUE TITLE
Can't seem to subscribe to the created openshift-pipeline-operator

### ISSUE THREAD

**Issue by brujoand** [openshift/tektoncd-pipeline-operator#94]
Sorry if this is just me not being very familiar with OLM, but I've ran into some problems here that I could use some help untangling.  

I've followed the readme, and gotten successfully through installing the prerequisites, setting up Minishift (both with virtualbox and hyperkit as the driver). Creating the development docker image went fine, and I also used ´minikube addons enable registry´ to setup a registry in my minikube cluster. Install OLM and running the UI works great. 

If I then try to follow the path of "Deploy openshift-pipelines-operator on minikube for testing" I run into problems because the openshift-operators namespace is used in the objects, but it's not the one we're told to create. I'm guessing this is because this PR #83 changed the objects, but the Readme is still the same. When adapting to this however I'm stuck at the event ´Error: cannot find volume "openshift-pipelines-operator-registry-configmap-server-toknd462" to mount into container "configmap-registry-server"´. This means I can't find the operator in the catalog.

If I instead go for the "Deploy openshift-pipelines-operator on minikube for testing" approach everything is deployed fine if I first create the namespace ´openshift-operators´, and I can even run ´./scripts/olm_catalog.sh > olm/openshift-pipelines-operator.resources.yaml´ after editing the image tag in  ´deploy/operator.yaml´. Using the namespace ´openshift-operators´ I can find the openshift-pipeline-operator but I can't subscribe because of the error ´Operator not available for selected namespaces´

If you can give me some feedback on my assumptions I'd be happy to update the readme, but most importantly what have I missed here that causes both deployment approaches to fail?

**Comment by nikhil-thomas** [comment]
>  ... the openshift-operators namespace is used in the objects, but it's not the one we're told to create. I'm guessing this is because this PR #83 changed the objects, but the Readme is still the same. ...

you are right. Read me should say `openshift-operators` instead of `openshift-pipelines-operator`. Thank  you for creating this issue. :+1:  PRs are welcome :)

**Comment by brujoand** [comment]
Sure thing 👍
I created #97 to update the readme, however as I'm still getting the ´Operator not available for selected namespaces´ when adding the subscription, I suspect I'm still missing a piece. Any insight on that?

**Comment by brujoand** [comment]
Just did a clean setup of this, and when I try to install OpenShift pipelines though the operator I can't. 
![Screenshot 2019-10-10 at 09 37 09](https://user-images.githubusercontent.com/124421/66548470-96634300-eb41-11e9-86d7-f7a989d0cf83.png)

´´´
kubectl get pods --all-namespaces
NAMESPACE             NAME                                          READY   STATUS    RESTARTS   AGE
kube-system           coredns-576cbf47c7-mm9fq                      1/1     Running   0          22m
kube-system           coredns-576cbf47c7-nk59j                      1/1     Running   0          22m
kube-system           etcd-minikube                                 1/1     Running   0          21m
kube-system           kube-addon-manager-minikube                   1/1     Running   0          21m
kube-system           kube-apiserver-minikube                       1/1     Running   0          21m
kube-system           kube-controller-manager-minikube              1/1     Running   0          21m
kube-system           kube-proxy-p9tj9                              1/1     Running   0          22m
kube-system           kube-scheduler-minikube                       1/1     Running   0          21m
kube-system           registry-ddx5l                                1/1     Running   0          22m
kube-system           registry-proxy-65s94                          1/1     Running   0          22m
kube-system           storage-provisioner                           1/1     Running   0          22m
olm                   catalog-operator-5cf9ff5855-ctf72             1/1     Running   0          14m
olm                   olm-operator-5d47954b9b-sqqrp                 1/1     Running   0          14m
olm                   operatorhubio-catalog-4csr9                   1/1     Running   0          14m
olm                   packageserver-645fbfcc6-6mnw9                 1/1     Running   0          14m
openshift-operators   openshift-pipelines-operator-registry-npd2z   1/1     Running   0          6m35s
´´´
Both ´tektoncd-pipeline-operator´ and ´operator-lifecycle-manager´ repos were synced today.
Any clues on debugging this?

**Comment by nikhil-thomas** [comment]
@brujoand are you creating the operatorsoruce manifest using the script `olm_catalog.sh`

**Comment by brujoand** [comment]
I had not as I hadn’t made any changes to the docker image. But retrying after generating the manifest gave the same error.

**Comment by nikhil-thomas** [comment]
is the openshift-pipelines-operator that we see in the pic above the already existing one or the one added by you

**Comment by nikhil-thomas** [comment]
in addition, there is a new workflow recommended by `operator-framework` to test operators locally
https://github.com/operator-framework/community-operators/blob/master/docs/testing-operators.md

**Comment by nikhil-thomas** [comment]
could you give a link to your CSV file

**Comment by nikhil-thomas** [comment]
what do you get when you do,
`oc get operatorgroup --all-namespaces`

**Comment by nikhil-thomas** [comment]
When `All namespaces` installation mode is selected OLM uses a predetermined Namespace to deploy the operator-deployment. The predetermined Namepsace is usually `openshift-operators`. We need to verify it is the same on your cluster.

ref: https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/operatorgroups.md (especially InstallModes and Supported OperatorGroups)

**Comment by brujoand** [comment]
The openshift-pipelines-operator was added by me, through ´kubectl apply -f olm/openshift-pipelines-operator.resources.yaml´ 

Not quite sure what you mean by CSV file, is it the yaml generated by olm_catalog.sh?

```
oc get operatorgroup --all-namespaces
NAMESPACE   NAME               AGE
olm         olm-operators      1h
operators   global-operators   1h
```
So I guess the last one should have been `openshift-operators`?

**Comment by brujoand** [comment]
Setting `NAMESPACE=operators` did the trick, thanks 👍

**Comment by nikhil-thomas** [comment]
in addition, checkout s https://github.com/operator-framework/community-operators/blob/master/docs/testing-operators.md
you might find it useful

**Comment by brujoand** [comment]
Thanks, I'm pretty new to openshift, olm and tekton so I need all the links I can get at this point :)

**Comment by nikhil-thomas** [comment]
I appreciate your efforts :+1: 
if you haven't joined already, join our slack: https://github.com/tektoncd/community/blob/master/contact.md#slack

**Comment by nikhil-thomas** [comment]
@brujoand shall we close this issue

**Comment by brujoand** [comment]
Yupp 👍
